powerauth-tpp-engine/src/main/java/io/getlime/security/powerauth/app/tppengine/repository/ConsentRepository.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.app.tppengine.repository;
+
+import io.getlime.security.powerauth.app.tppengine.repository.model.entity.ConsentEntity;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+/**
+ * Repository used for accessing the consent database.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Repository
+public interface ConsentRepository extends CrudRepository<ConsentEntity, String> {
+
+    @Query("SELECT c FROM ConsentEntity c WHERE c.id = :id ORDER BY c.version DESC")
```

Comment:
Again, my preference would be to use `consent_id` instead of `id`. But this is really a matter of style / philosophy.